Add autoconf, automake and libtool as dependencies

EgorBo opened the pull request:
Mono Runtime subset is enabled by default for `./build.sh` and it needs these packages on Linux.

EgorBo committed:
Add autoconf, automake and libtool as dependencies

EgorBo committed:
Add autoconf, automake and libtool as dependencies

jkotas commented:
We have the central list of dependencies here: https://github.com/dotnet/runtime/blob/master/docs/workflow/requirements/linux-requirements.md#toolchain-setup

It would be best to delete the redundant one from this doc.

EgorBo committed:
Merge branch 'master' of github.com:dotnet/runtime into fix-prereq

EgorBo committed:
Update instructions

EgorBo commented:
@jkotas @ViktorHofer Sorry for the delay, I've updated the docs:
1) Moved everything to `linux-requirements.md#toolchain-setup`
2) Removed notes regarding ARM, I guess it's already covered via clang-9 requirement
3) Removed `dnf`-based list - it's out of date

jkotas reviewed (approved):
Thanks
